### PR TITLE
New App - BBL Cricket

### PR DESCRIPTION
**Pull request by M0ntyP**
This app is for following your favourite BBL team (BBL = Big Bash League, Australian T20 cricket tournament).

Just select your team from the dropdown and it will show either the next upcoming game, the current game or previous game. 

For the upcoming game, you'll see a screen like this. Showing the team records and time of the next game, in your local timezone.
bbl_cricket

For a live game, you'll see something similar to this for the team batting first
bbl_cricket_live
And this for the chasing team
bbl_cricket_2nd
There is a slight difference in the information shown on the bottom row

And for a completed match, you'll see 
image

I had a lot of fun making this app especially dealing with all the nuances in the game of cricket. Please let me know if you find any issues as I'm sure there is a scenario I haven't allowed for 

Thanks!

### CODE REVIEW

File: apps/bblcricket/bbl_cricket.star [hunk 1]
```diff
@@ -0,0 +1,771 @@
...
+    "locality": "Adelaide",
+    "timezone": "Australia/Adelaide"
+}
+"""
+
+def main(config):
+    SelectedTeam = config.get("TeamList", DEFAULT_TEAM)
+    SelectedTeam = int(SelectedTeam)
+
+    Standings_JSON = http.get(Standings_URL).json()
```
Comment (matslina) [inline apps/bblcricket/bbl_cricket.star:32]: We'll need some caching around these calls to make sure we're not hitting the API to frequently.

File: apps/bblcricket/bbl_cricket.star [hunk 2]
```diff
@@ -0,0 +1,771 @@
...
+
+            location = DEFAULT_LOCATION
+            loc = json.decode(location)
+            timezone = loc["timezone"]
+
+            MyTime = time.parse_time(StartTime, format = "2006-01-02T15:04:00.000Z").in_location(timezone)
+            Time = MyTime.format("15:04")
+            Date = MyTime.format("Jan 2")
+
+            Standings_JSON = http.get(Standings_URL).json()
```
Comment (matslina) [inline apps/bblcricket/bbl_cricket.star:544]: Caching here too.

File: apps/bblcricket/bbl_cricket.star [hunk 3]
```diff
@@ -0,0 +1,771 @@
...
+    schema.Option(
+        display = "Hobart Hurricanes",
+        value = "4845",
+    ),
+    schema.Option(
+        display = "Perth Scorchers",
+        value = "4846",
+    ),
+    schema.Option(
+        display = "Melboure Renegades",
```
Comment (matslina) [inline apps/bblcricket/bbl_cricket.star:703]:
```suggestion
        display = "Melbourne Renegades",
```

**Review by matslina — CHANGES_REQUESTED** [review]
Thanks for the PR. I bet a lot of cricket fans will really enjoy this one.

A few changes needed before we can merge and publish. Primarily need some caching to keep outgoing request rates as low as possible.

File: apps/bblcricket/bbl_cricket.star (hunk 1 above)
Comment (M0ntyP) [reply apps/bblcricket/bbl_cricket.star:32]: Can do, I'll fix that up

File: apps/bblcricket/bbl_cricket.star (hunk 3 above)
Comment (M0ntyP) [reply apps/bblcricket/bbl_cricket.star:703]: I can't believe I spelt that wrong! I must have looked at the dropdown 100s of times during testing 

Great pick up!

**Comment by M0ntyP** [comment]
The spelling errors have been corrected. I have also implemented caching and will need to test it against a live game tonight to ensure that it works correctly

**Comment by M0ntyP** [comment]
I have uploaded a new version of the .star file now. Hopefully its good to go!

**Review by matslina — APPROVED** [review]

File: apps/bblcricket/bbl_cricket.star (hunk 2 above)
Comment (matslina) [reply apps/bblcricket/bbl_cricket.star:556]: This one should be a `get_cachable_data()`, right?

Comment (matslina) [reply apps/bblcricket/bbl_cricket.star:556]: Or just removed entirely since you're geting `Standings_JSON` above on L536?

Comment (M0ntyP) [reply apps/bblcricket/bbl_cricket.star:556]: Yes, we can remove it as its already defined earlier on

Comment (M0ntyP) [reply apps/bblcricket/bbl_cricket.star]: I've commented it out now

**Pull request merged**